CMakeLists.txt

```diff
@@ -88,7 +88,9 @@ find_package(Threads REQUIRED)
 MESSAGE(STATUS "Threads library: " ${CMAKE_THREAD_LIBS_INIT})
 
 set(Boost_NO_BOOST_CMAKE ON)
-set(CMAKE_CXX_STANDARD 11)
+if (NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 11)
+endif ()
 set(CMAKE_C_STANDARD 11)
```

Comment:
For options alignment, perhaps perform the same change for `CMAKE_C_STANDARD`?

Reply:
No. The C standard is only applied to
- The C headers (`*.h` files) under https://github.com/apache/pulsar-client-cpp/tree/main/include/pulsar/c
- The C examples, i.e. the `*.c` files under https://github.com/apache/pulsar-client-cpp/tree/main/examples

They are never affected by any external dependency.

We should still stick to the C++11 and C11 standard for this project, but we cannot avoid the minimum standard version requirements from external C++ dependencies.

Reply:
Thanks for your explanation! That sounds good to me.